cohttp-lwt/src/s.ml

```diff
@@ -29,6 +29,10 @@ module type Net = sig
   val close : IO.ic -> IO.oc -> unit
 end
 
+module type Sleep = sig
+  val sleep_ns : int64 -> unit Lwt.t
```

Comment:
If we are functorizing it this way, and we only need milliseconds, should we have `sleep_ms` here instead, so that it is also clear why it is different from mirage-time?

Reply:
I thought a bit about the necessary order of magnitude of the idle timeout. I believe sensible values range from days = 2¹⁶ s to about 50μs = 2⁻¹⁴s (roundtrip to localhost). This range barely fits into a 31bit integer, so I'm afraid we'll need a full `int64`. Now with 64bits order of magnitude won't matter that much. I would choose μs, but since `mirage-time` already uses ns, I would just stick with that and also expose that in the `Connection_cache`.

Reply:
Fair enough, if the range is so large I agree

Reply:
there seems to be a reason, why `Unix.sleep` expects `float`.

Reply:
No idea why is like that historically.

Afaik in glibc, `sleep` only works on seconds and you need `nanosleep` for fractions of seconds.
I don't know why in ocaml's `Unix` module this is reflected by `sleep` which takes an integer (in seconds) and `sleepf` which, instead, takes a float to allow for fractions of seconds. Maybe this is to work also on older 32 bits systems?
My guess is that in `Lwt_unix.sleep` they just wanted to remain close to the `Unix` module.